src/data/nat/enat.lean

```diff
@@ -17,10 +17,50 @@ namespace enat
 instance : has_zero enat := ⟨some 0⟩
 instance : has_one enat := ⟨some 1⟩
 instance : has_add enat := ⟨λ x y, ⟨x.dom ∧ y.dom, λ h, get x h.1 + get y h.2⟩⟩
-instance : has_coe ℕ enat := ⟨some⟩
-instance (n : ℕ) : decidable (n : enat).dom := is_true trivial
+instance : has_le enat := ⟨λ x y, ∃ h : y.dom → x.dom, ∀ hy : y.dom, x.get (h hy) ≤ y.get hy⟩
+instance : has_top enat := ⟨none⟩
+instance : has_bot enat := ⟨0⟩
+instance : has_sup enat := ⟨λ x y, ⟨x.dom ∧ y.dom, λ h, x.get h.1 ⊔ y.get h.2⟩⟩
+
+@[simp] lemma some_eq_coe (n : ℕ) :
+  (some n : enat) = (↑n : enat) :=
+  -- (@coe nat enat (@coe_to_lift nat enat (@coe_base nat enat nat.cast_coe)) n) = n :=
+begin
+ induction n with n IH, {refl},
+ simp only [nat.succ_eq_add_one, nat.cast_succ, IH.symm],
+ apply roption.ext', { show true ↔ (true ∧ true), simp },
+ intros h₁ h₂, refl
+end
+
+@[reducible] def finite (x : enat) : Prop := x.dom
```

Comment:
Why is this reducible?

Reply:
Frankly, because I copied the definition from `ring_theory/multiplicity.lean`. It should probably not have the attribute...